src/protocols/musig2/mod.rs

```diff
@@ -0,0 +1,167 @@
+#![allow(non_snake_case)]
+
+//! Simple ed25519
+//!
+//! See https://tools.ietf.org/html/rfc8032
+//! This is an implementation of the Musig2 protocol as shown in https://eprint.iacr.org/2020/1261.pdf with the addition named Musig2* suggested in Section B of the paper.
+//! We implement the v = 2 (NUMBER_OF_NONCES) version, meaning there are 2 nonces generated by each party.
+
+use super::{ExpandedKeyPair, Signature};
+use curv::cryptographic_primitives::hashing::DigestExt;
+use curv::elliptic::curves::{Ed25519, Point, Scalar};
+use rand::Rng;
+use sha2::{digest::Digest, Sha512};
+
+const NUMBER_OF_NONCES: usize = 2;
```

Comment:
This should probably be public, because you expose arrays of that length to the user